addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/rekognition/listeners/BaseAsyncResultListener.java

```diff
@@ -0,0 +1,115 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.rekognition.listeners;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.enrichment.EnrichmentMetadata;
+import org.nuxeo.ai.services.DocMetadataService;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.validation.DocumentValidationException;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.EventContext;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+public abstract class BaseAsyncResultListener implements PostCommitEventListener {
+
+    private static final Logger log = LogManager.getLogger(BaseAsyncResultListener.class);
+
+    public static final String JOB_ID_CTX_KEY = "jobId";
+
+    @Override
+    public void handleEvent(EventBundle eb) {
+        eb.forEach(this::handleEvent);
+    }
+
+    protected void handleEvent(Event event) {
+        EventContext ctx = event.getContext();
+        if (ctx == null) {
+            return;
+        }
+
+        String jobId = (String) ctx.getProperties().get(JOB_ID_CTX_KEY);
+        if (getFailureEventName().equals(event.getName())) {
+            log.info("Async Rekognition job failed for id " + jobId);
+            return;
+        }
+
+        byte[] bytes = getStore().get(jobId);
+        if (bytes == null) {
+            log.error("Could not find data for JobId " + jobId);
+            return;
+        }
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+             ObjectInputStream ois = new ObjectInputStream(bais)) {
+
+            @SuppressWarnings("unchecked")
+            Map<String, Serializable> params = (Map<String, Serializable>) ois.readObject();
+
+            Collection<EnrichmentMetadata> enrichment = getEnrichmentMetadata(jobId, params);
+            enrichment.forEach(this::saveMetadata);
+        } catch (IOException | ClassNotFoundException e) {
+            log.error(e);
+        }
+    }
+
+    protected void saveMetadata(EnrichmentMetadata metadata) {
```

Comment:
where is this used?

Reply:
are you assuming the customer wants to save the metadata result?

Reply:
it's a refactored part of code splitting your method. the method saves metadata into a doc. perhaps a better naming will help